stdlib/3/os/__init__.pyi

```diff
@@ -285,63 +356,133 @@ def fstat(fd: int) -> stat_result: ...
 def fstatvfs(fd: int) -> statvfs_result: ...  # Unix only
 def fsync(fd: int) -> None: ...
 def ftruncate(fd: int, length: int) -> None: ...  # Unix only
+if sys.version_info >= (3, 5):
+    def get_blocking(fd: int) -> bool: ...  # Unix only
+    def set_blocking(fd: int, blocking: bool) -> None: ...  # Unix only
 def isatty(fd: int) -> bool: ...  # Unix only
+if sys.version_info >= (3, 3):
+    def lockf(fd: int, cmd: int, length: int) -> None: ...  # Unix only
 def lseek(fd: int, pos: int, how: int) -> int: ...
-def open(file: _PathType, flags: int, mode: int = ...) -> int: ...
+if sys.version_info >= (3, 3):
+    def open(file: _PathType, flags: int, mode: Optional[int] = ..., *, dir_fd: Optional[int] = ...) -> int: ...
+else:
+    def open(file: _PathType, flags: int, mode: Optional[int] = ...) -> int: ...
 def openpty() -> Tuple[int, int]: ...  # some flavors of Unix
 def pipe() -> Tuple[int, int]: ...
+if sys.version_info >= (3, 3):
+    def pipe2(flags: int) -> Tuple[int, int]: ...  # some flavors of Unix
+    def posix_fallocate(fd: int, offset: int, length: int) -> None: ...  # Unix only
+    def posix_fadvise(fd: int, offset: int, length: int, advice: int) -> None: ...  # Unix only
+    def pread(fd: int, buffersize: int, offset: int) -> bytes: ...  # Unix only
+    def pwrite(fd: int, string: bytes, offset: int) -> int: ...  # Unix only
 def read(fd: int, n: int) -> bytes: ...
+if sys.version_info >= (3, 3):
+    # sendfile() has a different variation on FreeBSD and Mac OS X
+    def sendfile(out_fd: int, in_fd: int, offset: Optional[int], count: int) -> int: ...  # Unix only
+    def readv(fd: int, buffers: Sequence[bytearray]) -> int: ...  # Unix only
+    def writev(fd: int, buffers: Sequence[bytes]) -> int: ...  # Unix only
+
+    terminal_size = NamedTuple('terminal_size', [('columns', int), ('lines', int)])
+    def get_terminal_size(fd: Optional[int] = ...) -> terminal_size: ...
+
+if sys.version_info >= (3, 4):
+    def get_inheritable(fd: int) -> bool: ...
+    def set_inheritable(fd: int, inheritable: bool) -> None: ...
+
 def tcgetpgrp(fd: int) -> int: ...  # Unix only
 def tcsetpgrp(fd: int, pg: int) -> None: ...  # Unix only
 def ttyname(fd: int) -> str: ...  # Unix only
 def write(fd: int, string: bytes) -> int: ...
-def access(path: _PathType, mode: int) -> bool: ...
-def chdir(path: _PathType) -> None: ...
+if sys.version_info >= (3, 3):
+    def access(path: _FdOrPathType, mode: int, *, dir_fd: Optional[int] = ...,
+               effective_ids: Optional[bool] = ..., follow_symlinks: Optional[bool] = ...) -> bool: ...
```

Comment:
I think these two are supposed to be just bool, not `Optional[bool]`, though in reality it will accept anything and call `bool()` on it.